tests/test_samples.py

```diff
@@ -0,0 +1,54 @@
+"""Snapshot test of formatting of sample files.
+
+Code samples are located in tests/samples. To add a new sample create a new
+file and create a snapshot of the formatting result with
+
+    $ pytest --snapshot-update
+
+Commit both the sample as well as the updated snapshot.
+"""
+
+import io
+from pathlib import Path
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from testutils import zeekscript
+
+SAMPLES_DIR = Path("tests/samples")
+
+
+def _format(script: zeekscript.Script):
+    """Formats a given `Script`"""
+    buf = io.BytesIO()
+    script.format(buf)
+    return buf.getvalue().decode()
+
+
+def _get_samples():
+    """Helper to enumerate samples"""
+
+    # We exclude directories since we store snapshots along with the samples.
+    # This assumes that there are no tests in subdirectories of `SAMPLES_DIR`.
+    try:
+        return [sample for sample in SAMPLES_DIR.iterdir() if sample.is_file()]
+    except FileNotFoundError:
```

Comment:
Would drop that `except`, otherwise it shows just "skipped".

You could also make `_get_samples()` a generator instead of returning a new list it seems.

Reply:
The incorrect skipping should be fixed with correct computation of `SAMPLES_DIR` even in subdirs like you suggested in the previous item.

I tried turning `_get_samples` into a generator, but that seems to lead to the exception being raised in the scope where the generator is iterated, not in the generator scope. I think I'll just leave this as is.